apis/src/providers/api_requests.rs

```diff
@@ -77,9 +78,25 @@ impl ApiRequests {
     }
 
     pub fn challenge(&self, challenge_action: ChallengeAction) {
-        let msg = ClientRequest::Challenge(challenge_action);
-        self.websocket
-            .send(&serde_json::to_string(&msg).expect("Serde_json::to_string failed"));
+        let challenge_action = match challenge_action {
+            ChallengeAction::Create(details) => {
+                let auth_context = expect_context::<AuthContext>();
+                let account = match (auth_context.user)() {
+                    Some(Ok(Some(account))) => Some(account),
+                    _ => None,
+                };
+                let user = account.unwrap().user.username;
```

Comment:
unwrapping here shouldn't panic but just to be safe an if let Some(username) might be a better way to go